spec/services/order_syncer_spec.rb

```diff
@@ -213,7 +233,24 @@
   end
 
   describe "changing the ship address" do
-    let(:subscription) { create(:subscription, with_items: true, with_proxy_orders: true) }
+    let!(:distributor_address) { create(:address, address1: "Distributor Address") }
+    let!(:distributor) { create(:distributor_enterprise, address: distributor_address) }
+    let!(:original_bill_address) do
+      create(:address, firstname: "Walter", lastname: "Wolf", address1: "White")
+    end
+    let!(:original_ship_address) do
+      create(:address, firstname: "Melanie", lastname: "Miller", address1: "Magenta")
+    end
+    let!(:shipping_method) do
+      create(:shipping_method, distributors: [distributor], require_ship_address: true)
+    end
+
+    let(:subscription) do
+      create(:subscription, shop: distributor, bill_address: original_bill_address,
+                            ship_address: original_ship_address, shipping_method: shipping_method,
+                            with_items: true, with_proxy_orders: true)
+    end
+
```

Comment:
DRY
This is copy pasted from above, right?
I'd prefer to see this in a factory.

Reply:
Thanks for nudging me to do this, @luisramos0. Responding to this concern in a main comment, because this will need another review from the other reviewers.